official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,159 @@
+      self._train_time_batch_based = 0
+    else:  # For epoch_based
+      self._epochs = 0
+      self._train_time_epoch_based = 0
+
+  def on_batch_begin(self, batch, logs=None):
+    if self._batch_based:
+      self._time_start_batch_based = time.time()
+
+  def on_batch_end(self, batch, logs=None):
+    self._global_step += 1
+    if self._batch_based:
+      self._train_time_batch_based += time.time() - self._time_start_batch_based
+      examples_per_sec_batch_based = self._batch_size * (
+          self._global_step / self._train_time_batch_based)
+      self._logger.log_metric(
+          "examples_per_sec_batch_based",
+          examples_per_sec_batch_based,
+          global_step=self._global_step)
+
+  def on_epoch_begin(self, epoch, logs=None):
+    if self._epoch_based:
+      self._time_start_epoch_based = time.time()
+
+  def on_epoch_end(self, epoch, logs=None):
+    if self._epoch_based:
+      self._epochs += 1
+      self._train_time_epoch_based += time.time() - self._time_start_epoch_based
+      examples_per_sec_epoch_based = self._epoch_size * (
+          self._epochs / self._train_time_epoch_based)
+      self._logger.log_metric(
+          "examples_per_sec_epoch_based",
+          examples_per_sec_epoch_based,
+          global_step=self._global_step)
+
+
+class LoggingMetricCallback(tf.keras.callbacks.Callback):
+  """LoggingMetric callback.
+
+  By default, four metrics are logged: train accuracy, train loss, evaluation
+  accuracy and evaluation loss. Check _METRICS_TO_LOG for details.
+  """
+
+  def __init__(self, metrics=None, batch_based=None, epoch_based=None,
+               metric_logger=None):
+    if (batch_based is None) == (epoch_based is None):
+      raise ValueError("Exactly one of batch_based "
+                       "and epoch_based should be provided.")
+
+    self._batch_based = batch_based
+    self._epoch_based = epoch_based
+
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+    self._metrics = metrics or _METRICS_TO_LOG
+    for metric in self._metrics:
+      if metric.strip().lower() not in _METRICS_TO_LOG.keys():
+        raise ValueError("Unrecognized metric requested: {}".format(metric))
+
+  def on_train_begin(self, logs=None):
+    self._global_step = 0
+
+  def on_batch_end(self, batch, logs=None):
+    """Log metrics after each batch."""
+    self._global_step += 1
+    if self._batch_based:
+      for metric in self._metrics.keys():
+        # `val_acc` and `val_loss` can only be obtained after each epoch.
+        metric = metric.strip().lower()
+        if metric not in ["val_acc", "val_loss"]:
+          self._logger.log_metric(
+              _METRICS_TO_LOG[metric],
+              logs.get(metric),
+              global_step=self._global_step)
+
+  def on_epoch_end(self, epoch, logs=None):
+    """Log metrics after each epoch."""
+    if self._epoch_based:
+      for metric in self._metrics.keys():
+        metric = metric.strip().lower()
```

Comment:
This stripping and lowering is unnecessary once we just use the set of constants. Even if done, should be done on init, rather than every time we want to log.